medusa/clients/torrent/downloadstation.py

```diff
@@ -39,195 +36,135 @@ def __init__(self, host=None, username=None, password=None):
+            log.warning(self.message)
+            return False
 
-            return self.auth
 
     def _get_auth(self):
-        if self.session.cookies and self.auth:
-            return self.auth
-
-        params = {
-            'api': 'SYNO.API.Auth',
-            'version': 2,
-            'method': 'login',
-            'account': self.username,
-            'passwd': self.password,
-            'session': 'DownloadStation',
-            'format': 'cookie',
-        }
-
-        try:
-            self.response = self.session.get(self.urls['login'], params=params, verify=False)
-            self.response.raise_for_status()
-        except RequestException as error:
-            handle_requests_exception(error)
-            self.session.cookies.clear()
+        """Login to DownloadStation"""
+        errmap = {  400: 'No such account or incorrect password',
+                    401: 'Account disabled',
+                    402: 'Permission denied',
+                    403: '2-step verification code required',
+                    404: 'Failed to authenticate 2-step verification code'}
+        self.url = urljoin(self.host, 'webapi/auth.cgi')
+        params = {  'method': 'login',
+                    'api': 'SYNO.API.Auth',
+                    'version': 3,
+                    'session': 'DownloadStation',
+                    'account': self.username,
+                    'passwd': self.password}
+        self.auth = True
+        if not self._request(method='get', params=params):
             self.auth = False
-            return self.auth
+            return False
+        jdata = self.response.json()
+        if jdata.get('success'):
+            self.auth = True
         else:
-            return self._check_response()
+            err_code = jdata.get('error', {}).get('code', 100)
+            self.message = f"{self.name} login error: {errmap[err_code]}"
+            log.warning(self.message)
+            self.auth = False
+        return self.auth
 
     def _add_torrent_uri(self, result):
-
-        torrent_path = app.TORRENT_PATH
-
-        data = {
-            'api': 'SYNO.DownloadStation.Task',
-            'version': '1',
-            'method': 'create',
-            'session': 'DownloadStation',
-            'uri': result.url,
-        }
-
-        if not self._check_destination():
-            return False
-
-        if torrent_path:
-            data['destination'] = torrent_path
-        log.debug('Add torrent URI with data: {0}', json.dumps(data))
-        self._request(method='post', data=data)
-        return self._check_response()
+            # parameters "type' and "destination" must be coded as a json string so we add double quotes to those parameters
+        params = {  'api'        : 'SYNO.DownloadStation2.Task',
+                    'version'    : 2,
+                    'method'     : 'create',
+                    'create_list': 'false',
+                    'destination': f'"{app.TORRENT_PATH}"',
```

Comment:
As explained in the comment, the Download Station api expect some values to be formatted as a a json string so it is needed to add double quotes to them.
That's what the f-tring notation is doing

Reply:
I think it would be cleaner to have that being handled by the requests module.
To be sure you could also use json.dumps().
Like: 
```
        params = {
            'api': 'SYNO.DownloadStation2.Task',
            'version': 2,
            'method': 'create',
            'create_list': 'false',
            'destination': app.TORRENT_PATH,
            'type': 'url',
            'url': result.url
        }
        return self._add_torrent(json.dumps(params))
```
This is less error prone. And has a better readability. And should also work fine.
Normally you would use requests `json` param, to send json data in this case. That will stringify it under the hood.
like: `requests.post(webhook, json=payload, headers=headers)`. But the json param is not exposed through the _request() method.

Reply:
No that will not be possible, that will not be function on Download Station.
Please do not change this, I was problematic enough to get this working.

Reply:
How do you know, did you test this out?
If not, I can make the changes in a separate commit. If it works nice, if not, we just remove the commit? No harm no foul.
It would also not be that much work for you. But the current changes feel really hacky to me, so i'd like to avoid that if possible.